linkerd/app/core/src/serve.rs

```diff
@@ -79,3 +85,7 @@ where
         _ = shutdown => { Ok(()) }
     }
 }
+
+fn is_io(e: &(dyn std::error::Error + 'static)) -> bool {
+    e.is::<io::Error>() || e.source().map(is_io).unwrap_or(false)
+}
```

Comment:
TIOLI: this is recursive, it might be a little more efficient to write a loop, like
```suggestion
fn is_io(e: &(dyn std::error::Error + 'static)) -> bool {
    let mut error = Some(e);
    while let Some(e) = error {
        if e.is::<io::Error>() {
            return true;
        }
        error = e.source();
    }
    false
}
```
but, that's significantly less concise, and i imagine that error source chains aren't really deep enough for stack depth to matter here? your call.